alpine/FieldContainer.hpp

```diff
@@ -4,42 +4,64 @@
 #include <memory>
 #include "Manager/BaseManager.h"
 
-    // Define the FieldsContainer class
-    template <typename T, unsigned Dim = 3>
-    class FieldContainer{
-    
+// Define the FieldsContainer class
+template <typename T, unsigned Dim = 3>
+class FieldContainer{
     public:
         FieldContainer(Vector_t<double, Dim>& hr, Vector_t<double, Dim>& rmin,
                         Vector_t<double, Dim>& rmax, ippl::e_dim_tag decomp[Dim])
             : hr_m(hr), rmin_m(rmin), rmax_m(rmax) {
-            for (unsigned int i = 0; i < Dim; i++) {
-                decomp_m[i] = decomp[i];
+               for (unsigned int i = 0; i < Dim; i++) {
+                   decomp_m[i] = decomp[i];
+               }
             }
-     }
-    ~FieldContainer(){}
-    
-    VField_t<T, Dim> E_m;
-    Field_t<Dim> rho_m;
-    Vector_t<T, Dim> nr_m;
-
-    ippl::e_dim_tag decomp_m[Dim];
-
-    Vector_t<double, Dim> hr_m;
-    Vector_t<double, Dim> rmin_m;
-    Vector_t<double, Dim> rmax_m;
-    
-    std::shared_ptr<Mesh_t<Dim>> mesh_m;
-    std::shared_ptr<FieldLayout_t<Dim>> fl_m;
-    
-    // Access to the layout
-    FieldLayout_t<Dim>& getLayout() const { return *fl_m; }
-        
-    void initializeFields(std::shared_ptr<Mesh_t<Dim>> mesh, std::shared_ptr<FieldLayout_t<Dim>> fl) {
-        E_m.initialize(*mesh, *fl);
-        rho_m.initialize(*mesh, *fl);
-        fl_m = fl;
-        mesh_m = mesh;
-    }
-    };
+        ~FieldContainer(){}
+
+    private:
+        VField_t<T, Dim> E_m;
+        Field_t<Dim> rho_m;
+        ippl::e_dim_tag decomp_m[Dim];
+        Vector_t<double, Dim> hr_m;
+        Vector_t<double, Dim> rmin_m;
+        Vector_t<double, Dim> rmax_m;
+        std::shared_ptr<Mesh_t<Dim>> mesh_m;
+        std::shared_ptr<FieldLayout_t<Dim>> fl_m;
+
+    public:
+        VField_t<T, Dim>& getE() { return E_m; }
+        void setE(VField_t<T, Dim>& E) { E_m = E; }
```

Comment:
we should inline  all get/set functions

Reply:
Is this what you mean? see [347eeef](https://github.com/IPPL-framework/ippl/pull/240/commits/347eeefc541f2888f6a19c2130d77cb2a4a1d894)

```
    inline VField_t<T, Dim>& getE() { return E_m; }
    inline void setE(VField_t<T, Dim>& E) { E_m = E; }
```

Reply:
perfect